applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_truss_element_linear_3D2N.cpp

```diff
@@ -0,0 +1,203 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:     BSD License
+//           license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder
+//
+
+
+#include "adjoint_finite_difference_truss_element_linear_3D2N.h"
+#include "structural_mechanics_application_variables.h"
+#include "custom_response_functions/response_utilities/stress_response_definitions.h"
+#include "includes/checks.h"
+
+
+namespace Kratos
+{
+
+AdjointFiniteDifferenceTrussElementLinear::AdjointFiniteDifferenceTrussElementLinear(Element::Pointer pPrimalElement)
+    : AdjointFiniteDifferencingBaseElement(pPrimalElement)
+{
+}
+
+AdjointFiniteDifferenceTrussElementLinear::~AdjointFiniteDifferenceTrussElementLinear()
+{
+}
+
+void AdjointFiniteDifferenceTrussElementLinear::Calculate(const Variable<Vector >& rVariable,
+                        Vector& rOutput,
+                        const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    if(rVariable == STRESS_ON_GP)
+    {
+        TracedStressType traced_stress_type = static_cast<TracedStressType>(this->GetValue(TRACED_STRESS_TYPE));
+
+        const SizeType  GP_num = (mpPrimalElement->GetGeometry().IntegrationPoints()).size();
+        if (rOutput.size() != GP_num) 
+            rOutput.resize(GP_num, false);
+
+        switch (traced_stress_type)
+        {
+            case TracedStressType::FX:
+            {
+                std::vector< array_1d<double, 3 > > force_vector;
+                mpPrimalElement->GetValueOnIntegrationPoints(FORCE, force_vector, rCurrentProcessInfo);
+                for(IndexType i = 0; i < GP_num ; ++i)
+                    rOutput(i) = force_vector[i][0];
+                break;
+            }
+            default:
+                KRATOS_ERROR << "Invalid stress type! Stress type not supported for this element!" << std::endl;
+        }
+    }
+    else
+    {
+        rOutput.resize(1);
```

Comment:
is this needed?

Reply:
This is in line with the other adjoint elements. But I will keep this in mind and eventually change this for all adjoint elements.